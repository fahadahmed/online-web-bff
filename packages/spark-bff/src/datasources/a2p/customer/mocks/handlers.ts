import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
