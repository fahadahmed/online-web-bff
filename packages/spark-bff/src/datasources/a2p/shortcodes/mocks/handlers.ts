import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
