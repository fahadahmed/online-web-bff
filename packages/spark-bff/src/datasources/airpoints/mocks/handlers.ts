import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
