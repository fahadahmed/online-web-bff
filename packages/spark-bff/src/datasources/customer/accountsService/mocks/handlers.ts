import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
