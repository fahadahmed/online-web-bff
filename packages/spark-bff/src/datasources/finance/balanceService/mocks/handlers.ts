import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
