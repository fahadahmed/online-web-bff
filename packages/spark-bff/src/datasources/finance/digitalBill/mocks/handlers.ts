import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
