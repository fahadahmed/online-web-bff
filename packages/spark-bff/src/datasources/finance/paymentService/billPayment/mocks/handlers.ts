import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
