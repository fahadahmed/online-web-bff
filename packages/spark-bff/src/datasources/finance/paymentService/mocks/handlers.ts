import { rest } from 'msw';
import SuccessResponseMock from './register-card-success-response.json';
import FailedResponse from './register-card-failed-response.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const registerCardHandler = rest.post(
  `${baseUrl}/creditcard/:lineNumber`,
  async ({ params: { lineNumber } }, res, ctx) => {
    if (lineNumber === '0276404521') {
      return res(ctx.status(500), ctx.json(FailedResponse));
    }
    return res(ctx.status(200), ctx.json(SuccessResponseMock));
  },
);

const handlers = [registerCardHandler];

export default handlers;
