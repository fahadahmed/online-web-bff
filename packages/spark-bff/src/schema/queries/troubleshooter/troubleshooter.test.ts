import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import handlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';

beforeEach(() => {
  server.use(...handlers);
});

test('returns troubleshooting information', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        troubleshooter(
          lineNumber: "099991111"
          diagnosticCode: "internet_down-line_down-modem_can_not_reach"
          modemModel: "HG999b"
        ) {
          modemManufacturer
          modemImage
          modemSerialNumber
          diagnosticSummaryCode
          initialStep
          stepForModemModel
          helpSteps {
            stepId
            title
            bodyList {
              copy
              componentType
            }
            imageList {
              url
            }
            ctaTitle
            ctaList {
              label
              type
              webCTA
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.troubleshooter).toMatchSnapshot();
});

test('returns troubleshooting information when optional argument is missing', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        troubleshooter(
          lineNumber: "099991111"
          diagnosticCode: "internet_down-line_down-modem_can_not_reach"
        ) {
          modemManufacturer
          modemImage
          modemSerialNumber
          diagnosticSummaryCode
          initialStep
          stepForModemModel
          helpSteps {
            stepId
            title
            bodyList {
              copy
              componentType
            }
            imageList {
              url
            }
            ctaTitle
            ctaList {
              label
              type
              webCTA
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.troubleshooter).toMatchSnapshot();
});
