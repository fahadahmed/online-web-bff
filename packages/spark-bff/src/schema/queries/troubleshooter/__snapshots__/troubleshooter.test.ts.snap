// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns troubleshooting information 1`] = `
Object {
  "diagnosticSummaryCode": "internet_down-line_down-modem_can_not_reach",
  "helpSteps": Array [
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "We cannot connect to your modem.",
        },
        Object {
          "componentType": null,
          "copy": "According to our records, you have a Spark modem. If you have a different modem, it might explain why we can't connect.",
        },
        Object {
          "componentType": null,
          "copy": "Have a look at the details below and let us know if this is the modem you're using?",
        },
        Object {
          "componentType": "TROUBLESHOOT_MODEM_DETAILS",
          "copy": "Detected modem details for this line",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes, I am using this Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
        Object {
          "label": "Yes, I am using a different Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
        Object {
          "label": "No, I am using a non-Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
      ],
      "ctaTitle": "Are you using a Spark modem?",
      "imageList": Array [],
      "stepId": "check_blue_cable",
      "title": "Check modem cables",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The power cable is connected to the modem and switched on at the wall.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The phone cable is connected to the port labelled DSL and the other end is connected to the filter",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The filter is connected to the phone jack on the wall",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The telephone is connected to the filter",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The ethernet cable is connected to your computer and the LAN ethernet port on the modem",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Try using the internet.",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=success",
        },
        Object {
          "label": "No",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=restart_modem",
        },
      ],
      "ctaTitle": "Is your internet working?",
      "imageList": Array [
        Object {
          "url": "https://www.spark.co.nz/content/dam/telecomcms/content-images/myspark/connection-promise/Copper_connection-80.jpg",
        },
      ],
      "stepId": "check_modem_cable",
      "title": "Check modem cables",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "Restarting the modem can sometimes solve connection problems.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Switch your modem off at the wall or unplug it.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Wait for 30 seconds, then turn it  back on.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Restart your computer, wait two minutes, and then try using the internet.",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=success",
        },
        Object {
          "label": "No",
          "type": "CHAT",
          "webCTA": "#",
        },
      ],
      "ctaTitle": "Is your internet working?",
      "imageList": Array [
        Object {
          "url": "https://www.spark.co.nz/content/dam/telecomcms/content-images/myspark/connection-promise/modem_lg.PNG",
        },
      ],
      "stepId": "restart_modem",
      "title": "Restart modem",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "If you have any other problems with your connection, have a look at the Spark help section",
        },
        Object {
          "componentType": null,
          "copy": "Or wait 10 minutes and run the diagnostic tool again",
        },
      ],
      "ctaList": Array [],
      "ctaTitle": null,
      "imageList": null,
      "stepId": "success",
      "title": "Restart modem",
    },
  ],
  "initialStep": "check_blue_cable",
  "modemImage": "https://www.spark.co.nz/assets/spark/images/modems/hg999b.png",
  "modemManufacturer": "Huawei Technologies Co., Ltd.",
  "modemSerialNumber": "S8G8W14B88210024",
  "stepForModemModel": "HG999b",
}
`;

exports[`returns troubleshooting information when optional argument is missing 1`] = `
Object {
  "diagnosticSummaryCode": "internet_down-line_down-modem_can_not_reach",
  "helpSteps": Array [
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "We cannot connect to your modem.",
        },
        Object {
          "componentType": null,
          "copy": "According to our records, you have a Spark modem. If you have a different modem, it might explain why we can't connect.",
        },
        Object {
          "componentType": null,
          "copy": "Have a look at the details below and let us know if this is the modem you're using?",
        },
        Object {
          "componentType": "TROUBLESHOOT_MODEM_DETAILS",
          "copy": "Detected modem details for this line",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes, I am using this Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
        Object {
          "label": "Yes, I am using a different Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
        Object {
          "label": "No, I am using a non-Spark modem",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=check_modem_cable",
        },
      ],
      "ctaTitle": "Are you using a Spark modem?",
      "imageList": Array [],
      "stepId": "check_blue_cable",
      "title": "Check modem cables",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The power cable is connected to the modem and switched on at the wall.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The phone cable is connected to the port labelled DSL and the other end is connected to the filter",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The filter is connected to the phone jack on the wall",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The telephone is connected to the filter",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "The ethernet cable is connected to your computer and the LAN ethernet port on the modem",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Try using the internet.",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=success",
        },
        Object {
          "label": "No",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=restart_modem",
        },
      ],
      "ctaTitle": "Is your internet working?",
      "imageList": Array [
        Object {
          "url": "https://www.spark.co.nz/content/dam/telecomcms/content-images/myspark/connection-promise/Copper_connection-80.jpg",
        },
      ],
      "stepId": "check_modem_cable",
      "title": "Check modem cables",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "Restarting the modem can sometimes solve connection problems.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Switch your modem off at the wall or unplug it.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Wait for 30 seconds, then turn it  back on.",
        },
        Object {
          "componentType": "CHECKLIST_ITEM",
          "copy": "Restart your computer, wait two minutes, and then try using the internet.",
        },
      ],
      "ctaList": Array [
        Object {
          "label": "Yes",
          "type": "TROUBLESHOOTING_STEP",
          "webCTA": "www.spark.co.nz/myspark/diagnosticSummaryCode=internet_down-line_down-modem_can_not_reach&troubleshootingStep=success",
        },
        Object {
          "label": "No",
          "type": "CHAT",
          "webCTA": "#",
        },
      ],
      "ctaTitle": "Is your internet working?",
      "imageList": Array [
        Object {
          "url": "https://www.spark.co.nz/content/dam/telecomcms/content-images/myspark/connection-promise/modem_lg.PNG",
        },
      ],
      "stepId": "restart_modem",
      "title": "Restart modem",
    },
    Object {
      "bodyList": Array [
        Object {
          "componentType": null,
          "copy": "If you have any other problems with your connection, have a look at the Spark help section",
        },
        Object {
          "componentType": null,
          "copy": "Or wait 10 minutes and run the diagnostic tool again",
        },
      ],
      "ctaList": Array [],
      "ctaTitle": null,
      "imageList": null,
      "stepId": "success",
      "title": "Restart modem",
    },
  ],
  "initialStep": "check_blue_cable",
  "modemImage": "https://www.spark.co.nz/assets/spark/images/modems/hg999b.png",
  "modemManufacturer": "Huawei Technologies Co., Ltd.",
  "modemSerialNumber": "S8G8W14B88210024",
  "stepForModemModel": "HG999b",
}
`;
